### PR TITLE
Remove req from lsn map when committing.

### PR DESCRIPTION
**Pull request by xiaoxichen**

### CODE REVIEW

**Review by sanebay — APPROVED** [review]

**Comment by sanebay** [comment]
Did your test shows memory usage not increasing via grafana ? Whats the current usage ?

**Comment by xiaoxichen** [comment]
yes the homestore long running UT shows stable memory usage. Will continue to check on SM level.

File: src/lib/replication/repl_dev/raft_repl_dev.cpp [hunk 1]
```diff
@@ -758,6 +758,8 @@ void RaftReplDev::handle_commit(repl_req_ptr_t rreq, bool recovery) {
 
     // Remove the request from repl_key map.
     m_repl_key_req_map.erase(rreq->rkey());
+    // Remove the request from lsn map.
+    m_state_machine->unlink_lsn_to_req(rreq->lsn());
```
Comment (JacksonYao287) [inline src/lib/replication/repl_dev/raft_repl_dev.cpp:762]: if we do this , we will lose the entry cache in leader.
what`s the problem if remove this.
https://github.com/eBay/HomeStore/blob/72a355e80a8e8368b555637e491840bc6307eb77/src/lib/replication/repl_dev/raft_repl_dev.cpp#L1092

can the reqs in the leader be removed as those in follower when they are expired?

Comment (xiaoxichen) [reply src/lib/replication/repl_dev/raft_repl_dev.cpp:762]: this is not used anywhere except in `end_of_append_batch`  ATM.

Comment (xiaoxichen) [reply src/lib/replication/repl_dev/raft_repl_dev.cpp:762]: I think the `entry cache` is separated into two part - log and data.
Log is on logstore, as discussed in your logstore PR if the cache you add can be extend and cache last N records, regardless of flush.

Data is on the path of `RaftReplDev::on_fetch_data_received`, right now it just read from drive based on blkid. As https://github.com/eBay/HomeStore/issues/414 stated this approach is wrong after we introduce GC. If the fetch need to go through Index it goes through HO,  I think we may have a unified cache in HO which can serve customer IO as well.

**Review by JacksonYao287 — APPROVED** [review]

**Pull request merged**